core/templates/components/score-ring/score-ring.component.ts

```diff
@@ -16,26 +16,30 @@
  * @fileoverview Component for the animated score ring.
  */
 
-import { Component, Input, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { Component, Input, OnChanges, AfterViewInit, ViewChild, ElementRef, SimpleChanges } from '@angular/core';
 import { downgradeComponent } from '@angular/upgrade/static';
 import { QuestionPlayerConstants } from '../question-directives/question-player/question-player.constants';
 
 @Component({
   selector: 'oppia-score-ring',
   templateUrl: './score-ring.component.html'
 })
-export class ScoreRingComponent implements OnInit, OnChanges {
+export class ScoreRingComponent implements AfterViewInit, OnChanges {
   constructor() {}
   @Input() score;
   @Input() testIsPassed: boolean;
+  @ViewChild('scoreRing') scoreRingElement: ElementRef<SVGCircleElement>;
   circle: SVGCircleElement;
   radius: number;
   circumference: number;
   COLORS_FOR_PASS_FAIL_MODE = QuestionPlayerConstants.COLORS_FOR_PASS_FAIL_MODE;
 
   setScore(percent: number): void {
-    const offset = this.circumference - percent / 100 * this.circumference;
-    this.circle.style.strokeDashoffset = offset.toString();
+    let that = this;
+    setTimeout(function() {
+      const offset = that.circumference - percent / 100 * that.circumference;
+      that.circle.style.strokeDashoffset = offset.toString();
+    }, 2000);
```

Comment:
Just wondering, why do we need to `let that = this`?

Reply:
Inside a setTimeout call the keyword 'this' loses its meaning. If I tried to access the class properties using the `this` keyword, say for example `this.circumference`, it would throw an error, something like 'circumference is undefined'. This is because inside a setTimeout call, `this` points to the setTimeout call itself (which doesn't have a property called 'circumference'), not the class outside which actually has the circumference property we are trying to access.

A way around this is to save `this` into a variable, and use said variable wherever we would need to access the class. Of course, the variable could be named anything, I've named it 'that' for simplicity.

Reply:
As you are not using arrow function for setTimeout callback, `this` refers to the callback function whereas in case of arrow function, `this` refer to the class.

Reply:
I see, so changing it to an arrow function should do the trick?

Reply:
Thanks for the comment Ashutosh! I replaced it with an arrow function and it now works fine.